runtime/src/iree/hal/drivers/hip/graph_command_buffer.c

```diff
@@ -212,47 +302,66 @@ static void iree_hal_hip_graph_command_buffer_begin_debug_group(
+  iree_hal_hip_graph_command_buffer_t* command_buffer =
+      iree_hal_hip_graph_command_buffer_cast(base_command_buffer);
+
+  (void)command_buffer;
+  IREE_HIP_GRAPH_COMMAND_BUFFER_TRACE_ZONE_BEGIN_EXTERNAL(
+      command_buffer, location ? location->file.data : NULL,
+      location ? location->file.size : 0, location ? location->line : 0,
+      /*func_name=*/NULL, 0, label.data, label.size);
 }
 
 static void iree_hal_hip_graph_command_buffer_end_debug_group(
     iree_hal_command_buffer_t* base_command_buffer) {
-  // TODO: tracy event stack.
-}
-
-static iree_status_t iree_hal_hip_graph_command_buffer_execution_barrier(
-    iree_hal_command_buffer_t* base_command_buffer,
-    iree_hal_execution_stage_t source_stage_mask,
-    iree_hal_execution_stage_t target_stage_mask,
-    iree_hal_execution_barrier_flags_t flags,
-    iree_host_size_t memory_barrier_count,
-    const iree_hal_memory_barrier_t* memory_barriers,
-    iree_host_size_t buffer_barrier_count,
-    const iree_hal_buffer_barrier_t* buffer_barriers) {
   iree_hal_hip_graph_command_buffer_t* command_buffer =
       iree_hal_hip_graph_command_buffer_cast(base_command_buffer);
-  IREE_TRACE_ZONE_BEGIN(z0);
+  (void)command_buffer;
+  IREE_HIP_GRAPH_COMMAND_BUFFER_TRACE_ZONE_END(command_buffer);
+}
 
+static iree_status_t
+iree_hal_hip_graph_command_buffer_execution_barrier_internal(
+    iree_hal_hip_graph_command_buffer_t* command_buffer) {
   IREE_ASSERT_GT(command_buffer->graph_node_count, 0,
                  "expected at least one node before a barrier");
 
   // Use the last node as a barrier to avoid creating redundant empty nodes.
   if (IREE_LIKELY(command_buffer->graph_node_count == 1)) {
     command_buffer->hip_barrier_node = command_buffer->hip_graph_nodes[0];
     command_buffer->graph_node_count = 0;
-    IREE_TRACE_ZONE_END(z0);
     return iree_ok_status();
   }
 
-  IREE_HIP_RETURN_AND_END_ZONE_IF_ERROR(
-      z0, command_buffer->symbols,
+  IREE_HIP_RETURN_IF_ERROR(
+      command_buffer->symbols,
       hipGraphAddEmptyNode(
           &command_buffer->hip_barrier_node, command_buffer->hip_graph,
           command_buffer->hip_graph_nodes, command_buffer->graph_node_count),
       "hipGraphAddEmptyNode");
 
   command_buffer->graph_node_count = 0;
 
+  return iree_ok_status();
+}
+
+static iree_status_t iree_hal_hip_graph_command_buffer_execution_barrier(
+    iree_hal_command_buffer_t* base_command_buffer,
+    iree_hal_execution_stage_t source_stage_mask,
+    iree_hal_execution_stage_t target_stage_mask,
+    iree_hal_execution_barrier_flags_t flags,
+    iree_host_size_t memory_barrier_count,
+    const iree_hal_memory_barrier_t* memory_barriers,
+    iree_host_size_t buffer_barrier_count,
+    const iree_hal_buffer_barrier_t* buffer_barriers) {
+  iree_hal_hip_graph_command_buffer_t* command_buffer =
+      iree_hal_hip_graph_command_buffer_cast(base_command_buffer);
+  IREE_TRACE_ZONE_BEGIN(z0);
+
+  IREE_RETURN_AND_END_ZONE_IF_ERROR(
```

Comment:
Just do a normal `iree_status_t status =` and `return status` later.

Reply:
done